test_tableslurp.sh

```diff
@@ -59,45 +59,46 @@ UPDOWN_FILE="aws_s3_cp.txt"
 INDEX_FILE="index.txt"
 ORIG_TEST_FILE="orig_test_file.txt"
 CONTAINER="tester"
-SRC_LOCAL="${DIR_LOCAL}/${UPDOWN_FILE}"
-INDEX_SRC="${DIR_LOCAL}/${INDEX_FILE}"
+
+SRC_LOCAL_UPDOWN_FILE="${DIR_LOCAL}/${UPDOWN_FILE}"
+SRC_LOCAL_INDEX_FILE="${DIR_LOCAL}/${INDEX_FILE}"
+SRC_LOCAL_ORIG_TEST_FILE="${DIR_LOCAL}"/"${ORIG_TEST_FILE}"
+
 INDEX_BUCKET="${BUCKET}"
 BACKUP_BUCKET="${BUCKET}"
-ORIG_TEST_FILE_LOC="${DIR_LOCAL}"/"${ORIG_TEST_FILE}"
 
-# Create destinations on S3. Note: Dest for index file is at top-level,
-# i.e., not under a directory.
+# Create destinations on S3.
 # s3://kimlew-tablesnap-tests/tester:/tmp/tablesnap-test/test_aws_s3_cp.txt
-DEST_ON_S3=s3://"${BUCKET}"/"${CONTAINER}":"${SRC_LOCAL}"
-INDEX_DEST=s3://"${BUCKET}"/"${CONTAINER}":/"${INDEX_FILE}"
-
-
-echo "SRC_LOCAL is:" "${SRC_LOCAL}"
-echo "DEST_ON_S3 is:" "${DEST_ON_S3}"
-echo "INDEX_SRC is:" "${INDEX_SRC}"
-echo "INDEX_DEST is:" "${INDEX_DEST}"
+# s3://kimlew-tablesnap-tests/tmp/tablesnap-test/index.txt
+DEST_S3_UPDOWN_FILE=s3://"${BUCKET}"/"${CONTAINER}":"${SRC_LOCAL_UPDOWN_FILE}"
+DEST_S3_INDEX_FILE=s3://"${BUCKET}"/"${CONTAINER}":/"${INDEX_FILE}" # OR ${ SRC_LOCAL_INDEX_FILE}
+
+echo "SRC_LOCAL_UPDOWN_FILE is:" "${SRC_LOCAL_UPDOWN_FILE}"
+echo "DEST_S3_UPDOWN_FILE is:\t" "${DEST_S3_UPDOWN_FILE}"
+echo "SRC_LOCAL_INDEX_FILE is:" "${SRC_LOCAL_INDEX_FILE}"
+echo "DEST_S3_INDEX_FILE is:\t" "${DEST_S3_INDEX_FILE}"
 echo
 
-date > "${SRC_LOCAL}"
-echo "${SRC_LOCAL}" > "${INDEX_SRC}"
-echo "CHECK /tmp/tablesnap-test with ls after ${SRC_LOCAL} given content & $UPDOWN_FILE > index file:"
+date > "${SRC_LOCAL_UPDOWN_FILE}"
+echo "${SRC_LOCAL_UPDOWN_FILE}" > "${SRC_LOCAL_INDEX_FILE}"
+echo "CHECK /tmp/tablesnap-test with ls after file given content & is redirected to index.txt:"
 ls -lah "${DIR_LOCAL}"
-echo "Contents of INDEX_FILE: "
-cat "${INDEX_SRC}"
+echo "Contents of index.txt: "
+cat "${SRC_LOCAL_INDEX_FILE}"
 echo
 
-aws s3 cp "${SRC_LOCAL}" "${DEST_ON_S3}"
-aws s3 cp "${INDEX_SRC}" "${INDEX_DEST}"
-
+aws s3 cp "${SRC_LOCAL_UPDOWN_FILE}" "${DEST_S3_UPDOWN_FILE}"
+aws s3 cp "${SRC_LOCAL_INDEX_FILE}" "${DEST_S3_INDEX_FILE}"
+echo
 # Check if test file is IN /tmp/tablesnap-test.
 if aws s3 ls "${BUCKET}/${CONTAINER}:${DIR_LOCAL}"/ | grep "$UPDOWN_FILE"; then
 	echo "SUCCESS: UPDOWN_FILE has been copied to S3."
 else
 	echo "FAILURE: UPDOWN_FILE has NOT been copied to S3."
 fi
-
+echo
 # Check if index file is IN /tmp/tablesnap-test.
-if aws s3 ls "${BUCKET}/${INDEX_FILE}" | grep "$INDEX_FILE"; then
+if aws s3 ls "${DEST_S3_INDEX_FILE}" | grep "$INDEX_FILE"; then
 	echo "SUCCESS: INDEX_FILE has been copied to S3."
 else
 	echo "FAILURE: INDEX_FILE has NOT been copied to S3."
```

Comment:
I suggest calling `die` with a message here. If you can't upload the index file to S3, something has gone wrong and you shouldn't continue the test.

Reply:
Done.

Reply:
Looks good.